### DISCUSSION TITLE
Unexpected authorize 403

### DISCUSSION THREAD

**Discussion by joostvanhoof** [livewire/livewire#3418] [1. Help] [answered]
I have an issue where only one authorize method returns a 403, while all others work.

My component:

```php
use App\Models\Post;
use Livewire\Component;
use Illuminate\Foundation\Auth\Access\AuthorizesRequests;

class Example extends Component
{
	use AuthorizesRequests;

	public function updatePost($post_id)
	{
		$post = Post::findOrFail($post_id);

		$this->authorize('update', $post); // This passes like expected
	}

	public function createPost()
	{
		$this->authorize('create'); // This produces a 403
	}
}
```

And the Post policy:

```php
use App\Models\User;
use App\Models\Post;
use Illuminate\Auth\Access\HandlesAuthorization;

class PostPolicy
{
    use HandlesAuthorization;

    public function update(User $user, Post $post)
    {
        if ($user->can('edit posts')) {
            return $user->company_id === $post->company_id;
        }
    }

    public function create(User $user)
    {
        if ($user->can('create posts')) {
            return true;
        }
    }

 }
```

Now the weird thing is that if I do this in my blade view (the Livewire component) it passes as expected. I use the spatie permissions package but given that it passes here there doesn't seem to be anything wrong with the set roles / permissions.
```
@can('create posts')
  Test <-- this is shown
@endcan
```

So it looks like the permission is set the right way. Anything that I am missing?

**Comment by joshhanley** [comment, marked as answer]
@joostvanhoof you need to pass the post class into the create authorise method

```php
public function createPost()
{
    $this->authorize('create', Post::class);
}
```

Hope this helps!

**Reply by joostvanhoof** [reply]
Thanks man! Really appreciate the help. I didn't try this because the `create` method on the `PostPolicy` doesn't actually take the `Post` model as a parameter, but I guess this is to make sure the `authorize` method knows which model policy to look for?

**Reply by joshhanley** [reply]
No worries 🙂 yeah exactly. See it here https://laravel.com/docs/8.x/authorization#user-model-actions-that-dont-require-models
